Cargo.toml

```diff
@@ -43,6 +46,9 @@ regex = { version = "1.6.0", optional = true }
 time = { version = "0.3.14", features = ["formatting"], optional = true }
 paste = { version = "1.0.9", optional = true }
 bitvec = "1.0.1"
+rocksdb = { version = "0.18", default-features = false, features = ["multi-threaded-cf"] }
+bincode = { version = "2.0.0-rc.2" }
+uuid = { version = "1.1.2",  features = ["v4"] }
```

Comment:
Shouldn't these be optional and activated with the the `persistence` feature?

Reply:
Right now the persistence feature flag only "swaps" the Spine struct with the PersistentTrace struct so it gets used as part of all the other dbsp code:
https://github.com/vmware/database-stream-processor/pull/124/files#diff-f8789d5760a694bd6491c0ed6d25099ee9e3dcaed42f69d2f492e779b10c44b4R21

Without the persistence feature the persistent code will still compile and run the persistent tests that check for spine equivalence (which I figured might be nice during development to catch problems early). But if we want to change that so that we don't have any persistent code compiled in without that feature flag we can make them optional.

Maybe at least I should rename the feature to describe better what it does.